### PR TITLE
linux: add new APUv2/APUv3 gpio/leds driver modules

### PR DESCRIPTION
**Pull request by feckert**
Since kernel version 4.15 there is a new driver stack for APUv2 and APUv3 boards available.
We now use kernel version 5.4 on x86_64 targets. So we could use this new driver stack for APUv2 and APUv3 boards.

GPIO-Driver:
https://github.com/torvalds/linux/blob/v5.4/drivers/gpio/gpio-amd-fch.c

GPIO/LEDs Platform driver
https://github.com/torvalds/linux/blob/v5.4/drivers/platform/x86/pcengines-apuv2.c

### CODE REVIEW

File: target/linux/generic/hack-5.4/991-platform-x86-pcengines-apuv2-revert-simswitch.patch [hunk 1]
```diff
@@ -0,0 +1,32 @@
+From 8c9254d41881c81bea610193c6ac59c8cb8b79fe Mon Sep 17 00:00:00 2001
+From: Florian Eckert <fe@dev.tdt.de>
+Date: Fri, 27 Mar 2020 16:11:55 +0100
+Subject: [PATCH] Revert "platform/x86: pcengines-apuv2: wire up simswitch gpio
+ as led"
+
+This reverts commit 5037d4ddda31c2dbbb018109655f61054b1756dc.
```
Comment (hauke) [inline target/linux/generic/hack-5.4/991-platform-x86-pcengines-apuv2-revert-simswitch.patch:7]: Please add a description why you reverted this.
Just put the commit message also here.

**Review by hauke — APPROVED** [review]
I do not have the hardware, but this looks good to me.

**Comment by feckert** [comment]
@hauke thanks for your review. I have updated my pull request with your requested changes.

**Comment by hauke** [comment]
Thank you for the patches, they were applied to master

**Comment by pyther** [comment]
This version of the module doesn't support the APU4 platform. However, it seems like this branch added support. https://github.com/dvhart/linux-pdx86/blob/for-next/drivers/platform/x86/pcengines-apuv2.c#L210

**Comment by pyther** [comment]
The kernel 5.4 version of this module does not support the APU4 platform. Looks like it is supported starting in 5.5. https://github.com/torvalds/linux/blob/v5.4/drivers/platform/x86/pcengines-apuv2.c

**Comment by feckert** [comment]
@hauke thanks for the merge
@pyther we could backport this do openwrt.
The next release of openwrt is on kernel 5.4 so we will not get apu4 till we on 5.5 or later.

**Comment by pyther** [comment]
I've created a PR to backport support for the APUv4 platform. https://github.com/openwrt/openwrt/pull/3167

**Pull request closed without merging**